### PR TITLE
Prepare for er_diagram

### PR DESCRIPTION
**Pull request by AdmOfReality**

### CODE REVIEW

File: app/models/test.rb [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
+  has_many :questions
+  # has_and_belongs_to_many :users
+  has_many :tests_users
+  has_many :users, through: :tests_users
 
-  def self.sort_desc_test_name_by_category(title)
-    Test.joins("INNER JOIN categories ON tests.category_id = categories.id").where(categories: {title: title}).order(title: :desc).pluck(:title)
-  end
+  # def self.sort_desc_test_name_by_category(title)
+  #   Test.joins("INNER JOIN categories ON tests.category_id = categories.id").where(categories: {title: title}).order(title: :desc).pluck(:title)
```
Comment (SkiNy66) [inline app/models/test.rb]: Let's remove commented code

File: app/models/user.rb [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+  # has_and_belongs_to_many :tests
+  has_many :tests_users
+  has_many :tests, through: :tests_users
 
-  def show_list_tests_by_level(level)
-    Test.joins("INNER JOIN users on (tests.user_id = users.id)").where(tests: {level: level})
-  end
+  # def show_list_tests_by_level(level)
+  #   Test.joins("INNER JOIN users on (tests.user_id = users.id)").where(tests: {level: level})
+  # end
```
Comment (SkiNy66) [inline app/models/user.rb]: Same

File: db/migrate/20240311131018_create_tests_users_join_table.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CreateTestsUsersJoinTable < ActiveRecord::Migration[7.1]
+  def change
+    create_join_table :users, :tests do |t|
+      t.index [:user_id, :test_id],  unique: true
+      # t.index [:test_id, :user_id]
```
Comment (SkiNy66) [inline db/migrate/20240311131018_create_tests_users_join_table.rb]: same

File: app/models/test.rb (hunk 1 above)
Comment (AdmOfReality) [reply app/models/test.rb]: Removed all commented out unnecessary code

File: app/models/user.rb (hunk 2 above)
Comment (AdmOfReality) [reply app/models/user.rb]: Removed all commented out unnecessary code

File: db/migrate/20240311131018_create_tests_users_join_table.rb (hunk 3 above)
Comment (AdmOfReality) [reply db/migrate/20240311131018_create_tests_users_join_table.rb]: Removed all commented out unnecessary code

**Comment by AdmOfReality** [comment]
Поправил все замечания, связи обозначил, диаграмму подправил

**Review by SkiNy66 — APPROVED** [review]

**Pull request merged**